### ISSUE TITLE
Text zoom causes header menu to wrap, overlapping page content

### ISSUE THREAD

**Issue by joyously** [WordPress/wporg-mu-plugins#248]
I asked in two different Slack channels where to report this, and no one answered, so I'm opening this issue here.

While trying to read a [Handbook page](https://developer.wordpress.org/apis/handbook/filesystem/), the font size was too small for me, so I hit Ctrl+ to make it bigger. This made the menu in the fixed header overlap the rest of the header and the content. It seems to be handled differently when I expand the window to the right (perhaps with a resize handler), but the black background of the menu is still larger than the other black parts.
(I am using Firefox. My standard window size is about 1040px.)

![font-size](https://user-images.githubusercontent.com/7957680/185651640-75b0abcd-c2cf-4e11-b8ab-0b405b738c70.gif)

**Comment by ryelle** [comment]
To reproduce:

- In Firefox, enable "Zoom Text Only" under View -> Zoom
- Zoom in
- Eventually the menu will wrap to a second line

<img width="1178" alt="Screen Shot 2022-08-19 at 4 27 58 PM" src="https://user-images.githubusercontent.com/541093/185702021-ed889dfd-e6ac-4702-90f8-ce5d347a7b36.png">

Technical notes for whoever picks this up — the issue is that the text zoom in Firefox does not trigger the resize handler, which should collapse that second line into the "…" menu. This is why it works when the browser is resized, even after increasing the text size.

We should fix the overly large menu (try to get [this callback](https://github.com/WordPress/wporg-mu-plugins/blob/08352560ceeeb02c479a16a418e07ad74b2754ca/mu-plugins/blocks/global-header-footer/js/view.js#L162-L172) to run on text zoom too), and only worry about the little misalignment if it's easy to address (maybe by switching to em/rem for the spacing values here?)

<img width="494" alt="Screen Shot 2022-08-19 at 4 26 56 PM" src="https://user-images.githubusercontent.com/541093/185701828-a7aea670-b952-4e14-a9dc-8b5fb02f94bc.png">
